plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergParquetFileWriter.java

```diff
@@ -70,14 +73,21 @@ public IcebergParquetFileWriter(
                 trinoVersion,
                 Optional.empty(),
                 Optional.empty());
+        this.location = outputFile.location();
         this.metricsConfig = requireNonNull(metricsConfig, "metricsConfig is null");
-        this.inputFile = new ForwardingInputFile(fileSystem.newInputFile(outputFile.location()));
     }
 
     @Override
     public Metrics getMetrics()
     {
-        return fileMetrics(inputFile, metricsConfig);
+        ParquetMetadata parquetMetadata;
+        try {
+            parquetMetadata = new ParquetMetadataConverter().fromParquetMetadata(parquetFileWriter.getFileMetadata());
+        }
+        catch (IOException e) {
+            throw new TrinoException(ICEBERG_WRITER_OPEN_ERROR, format("Error creating Parquet file %s", location), e);
```

Comment:
ICEBERG_WRITER_OPEN_ERROR? "Error creating Parquet file"?
maybe the messge & code should be different here

Reply:
updated this